e2eTests/scenarios/PowerPlatform/runtest.ps1

```diff
@@ -0,0 +1,101 @@
+#  - Run the "CI/CD" workflow
+#  - Check that artifacts are created for the app and the PowerPlatform solution
+#  - Remove the repository
+#
+'@
+
+$errorActionPreference = "Stop"; $ProgressPreference = "SilentlyContinue"; Set-StrictMode -Version 2.0
+
+Remove-Module e2eTestHelper -ErrorAction SilentlyContinue
+Import-Module (Join-Path $PSScriptRoot "..\..\e2eTestHelper.psm1") -DisableNameChecking
+
+$branch = "main"
+$template = "https://github.com/$pteTemplate"
+$repoPath = (Get-Location).Path
+
+foreach($sourceRepo in @('bcsamples-takeorder', 'bcsamples-CoffeeMR','bcsamples-WarehouseHelper')) {
+    Push-Location
+    $repository = "$githubOwner/$repoName-$($sourceRepo.Split('-')[1])"
+
+    # Login
+    SetTokenAndRepository -github:$github -githubOwner $githubOwner -token $token -repository $repository
+
+    # Create repository1
+    CreateAlGoRepository `
+        -github:$github `
+        -template "https://github.com/microsoft/$sourceRepo" `
+        -repository $repository `
+        -branch $branch `
+        -contentScript {
+            Param([string] $path)
+            Remove-PropertiesFromJsonFile -path (Join-Path $path '.github/AL-Go-Settings.json') -properties @('environments','DeployTo*')
+        }
+
+    $repoPath = (Get-Location).Path
+    Write-Host "Repo Path: $repoPath"
+
+    $settings = Get-Content -Path '.github/AL-Go-Settings.json' -Raw | ConvertFrom-Json
+    Write-Host "PowerPlatform Solution Folder: $($settings.powerPlatformSolutionFolder)"
+
+    if ($settings.templateUrl -eq 'https://github.com/Microsoft/AL-Go-PTE@PPPreview') {
+        # Upgrade AL-Go System Files from PPPreview to main (PPPreview branch still uses Y/N prompt and doesn't support direct AL-Go development - i.e. freddydk/AL-Go@branch)
+        SetRepositorySecret -repository $repository -name 'GHTOKENWORKFLOW' -value $token
+        $parameters = @{
+            "templateUrl" = 'https://github.com/microsoft/AL-Go-PTE@main'
+            "directCommit" = 'Y'
+        }
+        RunWorkflow -name 'Update AL-Go System Files' -parameters $parameters -wait -branch $branch -repository $repository
+    }
+
+    # Upgrade AL-Go System Files to test version
+    RunUpdateAlGoSystemFiles -directCommit -wait -templateUrl $template -ghTokenWorkflow $token -repository $repository | Out-Null
+
+    CancelAllWorkflows -repository $repository
+
+    # Pull and test workflows
+    Pull
+    Test-Path -Path '.github/workflows/PullPowerPlatformChanges.yaml' | Should -Be $true -Because "PullPowerPlatformChanges workflow exists"
+    Test-Path -Path '.github/workflows/PushPowerPlatformChanges.yaml' | Should -Be $true -Because "PushPowerPlatformChanges workflow exists"
+    Test-Path -Path '.github/workflows/_BuildPowerPlatformSolution.yaml' | Should -Be $true -Because "_BuildPowerPlatformSolution workflow exists"
+
+    $run = RunCICD -repository $repository -branch $branch
+    WaitWorkflow -repository $repository -runid $run.id
```

Comment:
It looks like the tests are run independently for all the three repositories. Could they be run in parallel to optimize time?

Reply:
The time consuming part (waiting for CI/CD) will now be done after kicking off all repositories.